src/cargo/util/lints.rs

```diff
@@ -84,41 +126,32 @@ pub struct Lint {
     pub edition_lint_opts: Option<(Edition, LintLevel)>,
 }
 
-impl Lint {
-    pub fn level(&self, lints: &TomlToolLints, edition: Edition) -> LintLevel {
-        let edition_level = self
-            .edition_lint_opts
-            .filter(|(e, _)| edition >= *e)
-            .map(|(_, l)| l);
+impl LevelTrait for Lint {
```

Comment:
Couldn't this trait have just been function parameters?

Reply:
Yes, it could've; I switched it to being a function